crates/api_models/src/payments/additional_info.rs

```diff
@@ -0,0 +1,220 @@
+use common_utils::pii::{self};
+use masking::{
+    masked_string::{MaskedBankAccount, MaskedIban, MaskedRoutingNumber, MaskedSortCode},
+    Secret,
+};
+use utoipa::ToSchema;
+
+use crate::enums as api_enums;
+
+#[derive(Eq, PartialEq, Clone, Debug, serde::Deserialize, serde::Serialize, ToSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum BankDebitAdditionalData {
+    Ach(AchBankDebitAdditionalData),
+    Bacs(BacsBankDebitAdditionalData),
+    Becs(BecsBankDebitAdditionalData),
+    Sepa(SepaBankDebitAdditionalData),
+}
```

Comment:
```suggestion
pub enum BankDebitAdditionalData {
    Ach(Box<AchBankDebitAdditionalData>),
    Bacs(Box<BacsBankDebitAdditionalData>),
    Becs(Box<BecsBankDebitAdditionalData>),
    Sepa(Box<SepaBankDebitAdditionalData>),
}
```

With Boxing the enum variant the memory allocation will be efficient.